### ISSUE TITLE
Default value results in AttributeError

### ISSUE THREAD

**Issue by zya-codes** [GoogleCloudPlatform/dfcx-scrapi#11]
https://github.com/GoogleCloudPlatform/dfcx-scrapi/blob/57a97d416a15df8fd3470fdd217b153013a53b51/src/dfcx_scrapi/tools/dataframe_functions.py#L777-782

In function create_entity_from_dataframe
If `meta = None`, which is the default value, this throws the error: 

`AttributeError: 'NoneType' object has no attribute 'get'`

In function bulk_create_entity_from_dataframe, the docstring says that the required dataframe columns are `display_name, value, synonyms`. Does it also require meta?

**Comment by zya-codes** [comment]
One option would be to define the default meta value to be the defaults Dialogflow CX may use.

**Comment by kmaphoenix** [comment]
Good catch @zya-codes, it appears this was an oversight between the two functions.  

Semantically we should change [create_entity_from_dataframe](https://github.com/GoogleCloudPlatform/dfcx-scrapi/blob/57a97d416a15df8fd3470fdd217b153013a53b51/src/dfcx_scrapi/tools/dataframe_functions.py#L758) to `make_entity_pb_from_dataframe` or something similar, since it is technically not creating an Entity in Dialogflow CX at all but instead is making a protobuf of `types.EntityType`.   
And we should also mark this as an internal function since it's meant to be used in conjunction with `bulk_create_entity_from_dataframe`.

Re: the `meta` value, I think we can leave it as is but refactor the internal code to account for its existence or not.  
The only Required field is `kind` per the lower level library here: 
https://github.com/googleapis/python-dialogflow-cx/blob/main/google/cloud/dialogflowcx_v3beta1/types/entity_type.py#L68

So I think there are a few tasks to complete in order to make both of these functions work more smoothly:
- `create_entity_from_dataframe`
  - RE-namespace `create_entity_from_dataframe` to align with what it is actually doing (making a protobuf)
  - Add `kind` as a required arg to `create_entity_from_dataframe` and remove it from any reference with `meta`
  - Inside `create_entity_from_dataframe`, check for the existence of `meta` before attempting to get all of the attributes
- `bulk_create_entity_from_dataframe`
  - Add `kind` as a required column to `entities_df`

This should fulfill all both use cases where users required `meta` and also where it is not required, while still providing the minimum args needed to create the EntityType protobuf.

Will assign this to you for now, but please let me know if you have any questions on the above.

**Comment by kmaphoenix** [comment]
@Greenford @MRyderOC assigning to you both to discuss.  
An older request, but still valid.

I think the crux here revolves around two things:
1. name spacing our methods (should it be static? private? leading underscore? etc.) to avoid user confusion
2. architecture decisions on where certain methods should live

For (2), the DataFrame functions class was historically where we did "all things related to DataFrames", but that scope has drastically increased since the inception of this class.  
Now we have methods in this class that operate on Intent, Entity Type, and Transition Route Group resources all in one class.  
Additionally, there are some DataFrame methods in other Class like `intents.py`. 

I think this could _possibly_ lead to user confusion (i.e. "Where is the Intent DataFrame thing that does X?")

# Discuss
**_Should we consolidate methods into their semantically named classes (i.e. Intent DF stuff goes in intents.py, etc.) or should we simply leave things as-is?_**  

Feel free to chat internally and drop a TLDR here when available.
